core/trino-main/src/test/java/io/trino/execution/TestSqlTaskManager.java

```diff
@@ -264,6 +265,10 @@ public void testFailStuckSplitTasks()
         // Here we explicitly enqueue an indefinite running split runner
         taskExecutor.enqueueSplits(taskHandle, false, ImmutableList.of(mockSplitRunner));
         taskExecutor.start();
+        // wait for the task executor to start processing the split
+        while (!mockSplitRunner.isStarted()) {
+            Thread.sleep(1000);
```

Comment:
Also I assume it might be reasonable to have another future that would identify a split finished execution

Reply:
The split is never finished // interrupted. It is only terminated at taskExecutor.stop().

The unit test only verified that the sqlTaskManager could identify the task with a long-running split and mark the task to the failed state.  

It is because the sqlTaskManager issues failTask() for the task, which eventually issues a list of callback functions while doing the state transition of the task. In a real Trino cluster, one of which is responsible for interrupting the DriverSplitRunner(an implementation of SplitRunner), however, in the unit test, there is no such callback function. And it is another unit test that checks when doing state changes, the registered callback functions can be called successfully. 

Let me know if you have a suggestion for improvement!